hypervisor/boot/sbl/multiboot.c

```diff
@@ -116,6 +123,26 @@ static void *get_kernel_load_addr(void *kernel_src_addr)
 	return kernel_src_addr;
 }
 
+static void get_emmc_serial(char *cmdline, uint8_t *serial)
+{
+	char *arg;
+	char *param;
+	uint32_t len;
+
+	len = strnlen_s("androidboot.serialno=", MEM_2K);
+	arg = strstr_s(cmdline, MEM_2K, "androidboot.serialno=", len);
+
+	if (arg == NULL) {
+		memcpy_s(serial, MMC_SERIAL_LEN, "123456789abcde", MMC_SERIAL_LEN);
+		return;
+	}
+
+	param = arg + len;
+
+	memcpy_s(serial, MMC_SERIAL_LEN, param, MMC_SERIAL_LEN-1);
```

Comment:
btw, do you know if this serial # contains the f/w version info as you previously discovered?